helix-term/src/application.rs

```diff
@@ -32,19 +36,41 @@ pub type LspCallback =
 pub type LspCallbacks = FuturesUnordered<LspCallback>;
 pub type LspCallbackWrapper = Box<dyn FnOnce(&mut Editor, &mut Compositor) + Send>;
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Configuration {
+    pub theme: Option<String>,
+}
+
 pub struct Application {
     compositor: Compositor,
     editor: Editor,
 
+    themes: Arc<Themes>,
     callbacks: LspCallbacks,
 }
 
 impl Application {
-    pub fn new(mut args: Args) -> Result<Self, Error> {
+    pub fn new(
+        mut args: Args,
+        config: Configuration,
+        themes: Arc<Themes>,
+        config_loader: Arc<syntax::Loader>,
+    ) -> Result<Self, Error> {
         use helix_view::editor::Action;
         let mut compositor = Compositor::new()?;
         let size = compositor.size();
-        let mut editor = Editor::new(size);
+        let theme = if let Some(theme) = &config.theme {
+            if let Some(theme) = themes.get(theme) {
```

Comment:
Ditto here, also the `to_owned` branch is followed by a `clone` which is not ideal (it's a double clone)